### ISSUE TITLE
Installation of Vivid Store causes issues in Sitemap

### ISSUE THREAD

**Issue by skybluesofa** [VividWeb/vivid_store#29]
I was trying to add a new subpage in the sitemap and received this error:

```
{"error":
    {"type":"Whoops\\Exception\\ErrorException",
    "message":"Call to a member function canPublishPageTypeBeneathTarget() on boolean",
    "file":"\/Users\/daverogers\/Sites\/concrete5.7.2\/sub\/updates\/concrete5.7.3.1\/concrete\/src\/Page\/Type\/Type.php",
    "line":993,
    "trace":[{
        "file":"\/Users\/daverogers\/Sites\/concrete5.7.2\/sub\/updates\/concrete5.7.3.1\/concrete\/src\/Page\/Type\/Type.php",
        "line":993,
        "function":null,
        "class":null,
        "args":[]
    }]},
    "errors":["Call to a member function canPublishPageTypeBeneathTarget() on boolean"]
}
```

Steps I took to create the issue:
1. Go to Dashboard > Sitemap
2. Click the Home page
3. Click the 'Add Page' option
4. 500 error with the the above response

Steps I took to debug when the error was introduced to the system:
1. Uninstalled the last installed package, QuickBooks Payments, to see if that was the cause: Nope.
2. Uninstalled the next to last package, Vivid Store: Now I was able to add a new subpage
3. Re-installed Vivid Store. 500 error again.
4. Uninstalled Vivid Store: Can add a new subpage.

While I haven't looked into it much beyond that, I'm wondering if it's the 'Product Details' page or the 'Product Page' type that get installed at or around lines 67-101 in the {root}/packages/vivid_store/controller.php file.

**Comment by skybluesofa** [comment]
It seems like this might be a core issue. Looks like some columns in the db don't get updated correctly. I've put a post on the c5 forum at https://www.concrete5.org/community/forums/5-7-discussion/adding-a-new-page-type/#733441

I'll post an update if/when someone responds.

**Comment by brucewyne** [comment]
Came across this the other day as well. I answered on C5. Essentially, what's forgotten is the path to publish beneath. It's set within the model so it works, but saving in the dashboard page leaves you with an error if you don't select one of the radio buttons.

**Comment by skybluesofa** [comment]
It doesn't actually get saved, though. You'll see in this image that the ptPublishTargetTypeID is null.
![image](https://cloud.githubusercontent.com/assets/1657128/7288936/bbdf0eb2-e92e-11e4-97b4-756895e86715.png)

I responded in the forum, too. Basically that something is broken. Either there should be a default action if nothing is set, or it should be saved when the page type is added. I tend to think that the API should set the PublishTargetType to a default of 'All' (unless the default is overwritten when the page type is created).

As it is now, from an outsider's perspective, if this happens when I install the store, then it's the store that is broken. In reality, though, it's not the store that is broken, but the API.

**Comment by brucewyne** [comment]
Merged the pull request #30 that resolves this issue. Thank you so much for catching my oversight @skybluesofa
